registration/include/pcl/registration/impl/sample_consensus_prerejective.hpp

```diff
@@ -261,32 +276,30 @@ SampleConsensusPrerejective<PointSource, PointTarget, FeatureT>::computeTransfor
       continue;
     }
 
-    // Estimate the transform from the correspondences, write to transformation_
+    // Estimate the transform from the correspondences, write to transformation
+    Eigen::Matrix4f transformation;
     transformation_estimation_->estimateRigidTransformation(
-        *input_, sample_indices, *target_, corresponding_indices, transformation_);
+    	*input_, sample_indices, *target_, corresponding_indices, transformation);
 
-    // Take a backup of previous result
-    const Matrix4 final_transformation_prev = final_transformation_;
-
-    // Set final result to current transformation
-    final_transformation_ = transformation_;
-
-    // Transform the input and compute the error (uses input_ and final_transformation_)
-    getFitness(inliers, error);
-
-    // Restore previous result
-    final_transformation_ = final_transformation_prev;
+    // Transform the input and compute the error (uses input_ and transformation)
+    float error;
+    getFitness(transformation, inliers, error);
 
     // If the new fit is better, update results
-    inlier_fraction =
-        static_cast<float>(inliers.size()) / static_cast<float>(input_->size());
+    const float inlier_fraction =
+        static_cast<float> (inliers.size ()) / static_cast<float> (input_->size ());
 
     // Update result if pose hypothesis is better
-    if (inlier_fraction >= inlier_fraction_ && error < lowest_error) {
-      inliers_ = inliers;
-      lowest_error = error;
-      converged_ = true;
-      final_transformation_ = transformation_;
+    #pragma omp critical
+    {
+      if (inlier_fraction >= inlier_fraction_ && error < lowest_error)
```

Comment:
`inlier_fraction_` does not change, correct? Then the first condition does not require the `omp critical` and you could move it in a second if statement _around_ the `omp critical`, so that the critical region is only entered if this first condition is true.

Reply:
Done